### DISCUSSION TITLE
ImGui Log/Console for Vulkan renderer integration

### DISCUSSION THREAD

**Discussion by johantutak** [ocornut/imgui#6187] [New Users Build/Link/Run issues ONLY!]
Hey i'm currently struggling to create a log/console that integrates with my spdlog based assert system and would like if someone could point me in the correct direction or give me some help? I'm new to Vulkan, imgui and programming in generall.

My plan is for example to get printf, or cout and cin functionallity into imgui log/console!

**Comment by PathogenDavid** [comment]
Hello!

For future reference, the discussions section is [only for questions related to compiling/running Dear ImGui and loading font files](https://github.com/ocornut/imgui/discussions/4497). Future questions should be [opened as issues](https://github.com/ocornut/imgui/issues/new/choose).

Your question is too open-ended for any concrete advice, but I can point you towards some examples you can look at:

There is a basic console in the official examples. It can be accessed from the demo window under Examples > Console. The implementation lives in [`ExampleAppConsole`](https://github.com/ocornut/imgui/blob/e9743d85ddc0986c6babaa01fd9e641a47b282f3/imgui_demo.cpp#L6523-L6884).

There is also Dear ImGui's own log window. It's much simpler and can be accessed from the demo winder under Tools > Debug Log. It's implemented in [`ImGui::ShowDebugLogWindow`](https://github.com/ocornut/imgui/blob/e9743d85ddc0986c6babaa01fd9e641a47b282f3/imgui.cpp#L14207-L14266).

[This project](https://github.com/ocornut/imgui/issues/1902#issuecomment-401454648) has a log viewer that is integrated in spdlog. I didn't look at it closely but it appears [the bulk of the implementation lives here](https://github.com/abdes/asap_app_imgui/blob/afa431ec5da36de9cbd4743a01d8990db84928c4/main/src/ui/log/sink.cpp). (I found this by [searching `spdlog` in our GitHub issues](https://github.com/ocornut/imgui/issues?q=is%3Aissue+spdlog+is%3Aopen).)

-----------

As an aside, being new to programming in general and Vulkan is a recipe for suffering. Unless Vulkan just really vibes with you I'd strongly suggest sticking to OpenGL or Direct3D 11 until you're more stable on your feet.

**Reply by johantutak** [reply]
Thank you for the advice. I'll check it out and I'll think about it next time!
